uClinux-dist/freeswan/lib/version.3
.TH IPSEC_VERSION 3 "21 Nov 2001"
.\" RCSID $Id: version.3,v 1.1.1.1 2004/07/19 09:22:52 lgsoft Exp $
.SH NAME
ipsec ipsec_version_code \- get IPsec version code
.br
ipsec ipsec_version_string \- get full IPsec version string
.br
ipsec ipsec_copyright_notice \- get IPsec copyright notice
.SH SYNOPSIS
.B "#include <freeswan.h>
.sp
.B "const char *ipsec_version_code(void);"
.br
.B "const char *ipsec_version_string(void);"
.br
.B "const char **ipsec_copyright_notice(void);"
.SH DESCRIPTION
These functions provide information on version numbering and copyright
of the Linux FreeS/WAN IPsec implementation.
.PP
.I Ipsec_version_code
returns a pointer to a string constant
containing the current IPsec version code,
such as ``1.92'' or ``snap2001Nov19b''.
.PP
.I Ipsec_version_string
returns a pointer to a string constant giving a full version identification,
consisting of the version code preceded by a prefix identifying the software,
e.g. ``Linux FreeS/WAN 1.92''.
.PP
.I Ipsec_copyright_notice
returns a pointer to a vector of pointers,
terminated by a
.BR NULL ,
which is the text of a suitable copyright notice.
Each pointer points to a string constant (possibly empty) which is one line
of the somewhat-verbose copyright notice.
The strings are NUL-terminated and do not contain a newline;
supplying suitable line termination for the output device is
the caller's responsibility.
.SH SEE ALSO
ipsec(8)
.SH HISTORY
Written for the FreeS/WAN project by Henry Spencer.


uClinux-dist/release_notes
Release Notes for uClinux for Blackfin Version 2005R4

Version: 2005R4

Date: Dec. 30, 2005
Development Environments

Tool Chain: Toolchain Release 2005R4

u-boot: uboot_1.1.3 Release 2005R4

Host platform: Suse Linux 9.2 or above

Target board: STAMP Board

Note: However, other similar host platforms are also supported, but they are not well tested by now.
Deliverables

Source files

uClinux_dist_2005R4.tar.bz2

Linux ELF file

linux-bf533
linux-bf537
linux-bf561

Compressed Linux image

uImage-bf533
uImage-bf537
uImage-bf561

This document

release_notes_2005R4.pdf

Compressed archive of test results

test_results_bf533_2005R4.tar.gz
test_results_bf537_2005R4.tar.gz
test_results_bf561_2005R4.tar.gz

Summary of test results

test_results_summary_2005R4

Fixes in release 2005R4

See tracker at:

http://blackfin.uclinux.org/tracker/index.php?group_id=17&atid=141
Improvements

   1.
      Support BF561-EZKIT and add a tool to load code to run in core B.
   2.
      Support Bluetechnix CM Platform, which is also based on BF533.
   3.
      Enable module support in default configuration and generate module dependency file for modprobe.
   4.
      Upgrade to mainline source code
          *
            Upgrade uClibc to version 0.9.27.
          *
            Upgrade Linphone to version 1.1.0
          *
            Upgrade LTP to 20051205 release.
   5.
      Reorganize the scheduling and signal handling.
          *
            Defer scheduling and signal handling from different interrupt handlers to interrupt level 15.
          *
            Deal with the deferred exception handler and the defered interrupt handler in the same level of system call handler.
          *
            Run schedule and do signal from each interrupt handler above level 14.
          *
            Move softirq handler to interrupt level 14
   6.
      Prevent double fault exception.
          *
            Make sure that in an exception handler, we use a stack that is covered by a CPLB.
          *
            For exceptions other than CPLB misses, lower our priority to IRQ5, where we can use the normal kernel stack and use other regular kernel functions.
   7.
      Port new applications, libraries and tools
          *
            samba
          *
            strace
          *
            SQLite
          *
            lmbench3.
          *
            whetstone.
          *
            network audio player
          *
            ncurses
   8.
      Work around new hardware anomalys.
          *
            Bypass the first DCLPB fault exception, only handle the second one if it is from the same fault address.
          *
            Do not put SSYNC,CSYNC or IDLE in the second to last instruction of a hardware loop.
          *
            Do not use DMA_RUN bit to detect completion of a DMA transfer.
   9.
      Change paths from bfinnommu to blackfin, per request from Linux Kernel Mailing list.
  10.
      Enable oprofiler support in kernel and Port oprofiler tools
  11.
      Enable interrupt latency benchmark
  12.
      Add new device drivers
          *
            Enable IDE driver
          *
            Enable 802.11 WLAN driver
          *
            Enable PPI driver
          *
            Enable NAND flash driver
          *
            Port YAFFS driver
          *
            Add a common SPI channel driver. All high level drivers access SPI device by this driver.
  13.
      ADI1836 audio driver improvement
          *
            Reduce dummy buffer size from 256 bytes to 4 bytes
          *
            No interrupts generated in dummy DMA transfer
          *
            When audio driver is not active, DMA is also stopped.
          *
            Enable I2S mode for audio input and output. Modify mixer for I2S mode.
          *
            Enable multichannel output. Add a mixer function to change the output channels when play stereo.
          *
            Enable play multi independent audio streams concurrently.
          *
            Make audio DMA stable.
  14.
      Kernel debugging improvement
          *
            Enable hardware single stepping for application debugging.
          *
            Report precise hardware error information
          *
            debug kernel start. Install a mini-execption handler (replacing the U-Boot one), turn the UART on, and dump out the SEQSTAT register.
          *
            Enable hardware trace buffer in dump
  15.
      Improve power management
          *
            Implement the suspend function in bf533_pm_suspend.
          *
            SDRAM enters self-refresh mode before deepsleep to save more power.
          *
            Enable PM on BF537
  16.
      Enable PPPoE
  17.
      Enable DMA in SPI driver.
  18.
      Add board specific Kconfig Option for PF port/pin, which is used for flash and ethernet sharing.
  19.
      Enhance msh to support funtions in scripts.
  20.
      Transmite bfin atomic testandset operation in exception 4 into a new syscall, sys_bfin_spinlock.

Known Issues

A full list of known issues can be found at:

http://blackfin.uclinux.org/tracker/index.php?group_id=17&atid=141
No 	Issue Title
891	tcp stress testing echo service
900	busybox/msh,cd,tar and pipes
960	NFS unstable under heavy load
991	Infrequent audio crash
1012	After running dpmc to_deepsleep testcase, the kernel would panic after a while
1014	kernel built with audio configuration sometimes would fail to boot up and hangs

There are also some issues in the LTP test cases. They are recorded as bug 532, 561, 567, 732, 735, 742, 744, 745, 890 and 1010.
Build Procedure

1. Install Toolchain Release 2005R4

Go to http://blackfin.uclinux.org/frs/?group_id=18 for more information


uClinux-dist/user/mawk/version.c

/********************************************
version.c
copyright 1991-95.  Michael D. Brennan

This is a source file for mawk, an implementation of
the AWK programming language.

Mawk is distributed without warranty under the terms of
the GNU General Public License, version 2, 1991.
********************************************/

/*$Log: version.c,v $
/*Revision 1.2  2005/07/25 04:03:15  magicyang
/*New user folder check in
/*
 *Revision 1.10  1996/07/28 21:47:07  mike
 *gnuish patch
 *
 * Revision 1.9  1996/02/01  04:44:15  mike
 * roll a beta version
 *
 * Revision 1.8  1995/08/20  17:40:45  mike
 * changed _stackavail to stackavail for MSC
 *
 * Revision 1.7  1995/06/10  17:04:10  mike
 * "largest field" replaced by "max NF"
 *
*/

#include "mawk.h"
#include "patchlev.h"

static char mawkid[] = MAWK_ID ;

#define	 VERSION_STRING	 \
  "mawk 1.3%s%s %s, Copyright (C) Michael D. Brennan\n\n"

/* If use different command line syntax for MSDOS
   mark that in VERSION	 */

#ifndef DOS_STRING
#if  MSDOS && ! HAVE_REARGV
#define DOS_STRING  "MsDOS"
#endif
#endif

#ifndef DOS_STRING
#define DOS_STRING	""
#endif

static char fmt[] = "%-14s%10lu\n" ;

/* print VERSION and exit */
void
print_version()
{

   printf(VERSION_STRING, PATCH_STRING, DOS_STRING, DATE_STRING) ;
   fflush(stdout) ;

   print_compiler_id() ;
   fprintf(stderr, "compiled limits:\n") ;
   fprintf(stderr, fmt, "max NF", (long) MAX_FIELD) ;
   fprintf(stderr, fmt, "sprintf buffer", (long) SPRINTF_SZ) ;
   print_aux_limits() ;
   exit(0) ;
}


/*
  Extra info for MSDOS.	 This code contributed by
  Ben Myers
*/

#ifdef __TURBOC__
#include <alloc.h>		/* coreleft() */
#define	 BORL
#endif

#ifdef __BORLANDC__
#include <alloc.h>		/* coreleft() */
#define	 BORL
#endif

#ifdef	BORL
extern unsigned _stklen = 16 * 1024U ;
 /*  4K of stack is enough for a user function call
       nesting depth of 75 so this is enough for 300 */
#endif

#ifdef _MSC_VER
#include <malloc.h>
#endif

#ifdef __ZTC__
#include <dos.h>		/* _chkstack */
#endif


int
print_compiler_id()
{

#ifdef	__TURBOC__
   fprintf(stderr, "MsDOS Turbo C++ %d.%d\n",
	   __TURBOC__ >> 8, __TURBOC__ & 0xff) ;
#endif

#ifdef __BORLANDC__
   fprintf(stderr, "MS-DOS Borland C++ __BORLANDC__ %x\n",
	   __BORLANDC__) ;
#endif

#ifdef _MSC_VER
   fprintf(stderr, "Microsoft C/C++ _MSC_VER %u\n", _MSC_VER) ;
#endif

#ifdef __ZTC__
   fprintf(stderr, "MS-DOS Zortech C++ __ZTC__ %x\n", __ZTC__) ;
#endif

   return 0 ;			 /*shut up */
}


int
print_aux_limits()
{
#ifdef BORL
   extern unsigned _stklen ;
   fprintf(stderr, fmt, "stack size", (unsigned long) _stklen) ;
   fprintf(stderr, fmt, "heap size", (unsigned long) coreleft()) ;
#endif

#ifdef _MSC_VER
   fprintf(stderr, fmt, "stack size", (unsigned long) stackavail()) ;
#endif

#ifdef __ZTC__
/* large memory model only with ztc */
   fprintf(stderr, fmt, "stack size??", (unsigned long) _chkstack()) ;
   fprintf(stderr, fmt, "heap size", farcoreleft()) ;
#endif

   return 0 ;
}
